packages/fxa-settings/src/components/UnitRowSecondaryEmail/index.tsx

```diff
@@ -2,55 +2,159 @@
-    {
-      onError: (error) => {
-        sentryMetrics.captureException(error);
-        revealAlertBar();
-      },
-      onCompleted: () => {
-        navigate('/beta/settings/emails/verify', { state: { email } });
-      },
-    }
+  const showSuccess = useCallback(
+    (message: string) => {
+      setAlertMessage([message, 'success']);
+      revealAlertBar();
+    },
+    [setAlertMessage, revealAlertBar]
+  );
+
+  const showError = useCallback(
+    (message: string) => {
+      setAlertMessage([message, 'error']);
+      revealAlertBar();
+    },
+    [setAlertMessage, revealAlertBar]
+  );
+
+  const [resendEmailCode] = useHandledMutation(RESEND_EMAIL_CODE_MUTATION, {
+    onCompleted() {
+      navigate('/beta/settings/emails/verify', { state: { email } });
+    },
+    onError(error) {
+      showError(`Sorry, there was a problem re-sending the verification code.`);
+      throw error;
+    },
+  });
+
+  const [makeEmailPrimary] = useHandledMutation(MAKE_EMAIL_PRIMARY_MUTATION, {
+    onCompleted() {
+      showSuccess(`${email} is now your primary email.`);
+    },
+    onError(error) {
+      showError(`Sorry, there was a problem changing your primary email.`);
+      throw error;
+    },
+    update: (cache) => {
+      cache.modify({
+        fields: {
+          account: (existing: Account) => {
+            const account = thaw(existing);
+            account.emails.find((m) => m.email === email)!.isPrimary = true;
+            account.emails.find(
+              (m) => m.isPrimary && m.email !== email
+            )!.isPrimary = false;
+            return account;
+          },
+        },
+      });
+    },
+  });
+
+  const [deleteEmail] = useHandledMutation(DELETE_EMAIL_MUTATION, {
+    onCompleted() {
+      showSuccess(`${email} email successfully deleted.`);
+    },
+    onError(error) {
+      showError(`Sorry, there was a problem deleting this email.`);
+      throw error;
+    },
+    update: (cache) => {
+      cache.modify({
+        fields: {
+          account: (existing: Account) => {
+            const account = thaw(existing);
+            account.emails.splice(
+              account.emails.findIndex((m) => m.email === email),
+              1
+            );
+            return account;
+          },
+        },
+      });
+    },
+  });
+
+  const SecondaryEmailUtilities = ({ children }: { children: ReactNode }) => (
+    <>
+      {verifiedAction && !verified && (
+        <ModalVerifySession
+          onDismiss={() => {
+            showError(
+              `Sorry, you'll need to verify your current session to perform this action.`
+            );
+          }}
+          onError={(error) => {
+            showError(error.message);
+          }}
+          onCompleted={verifiedAction}
+        />
+      )}
+      {alertBarRevealed && alertMessage && (
+        <AlertBar onDismiss={hideAlertBar} type={alertMessage[1]}>
+          <p data-testid={`alert-bar-message-${alertMessage[1]}`}>
```

Comment:
This is just a [tuple](https://www.typescriptlang.org/docs/handbook/basic-types.html#tuple), pretty standard but I can change it to an object. If we were on TypeScript 4 we could label the elements, but alas.

Reply:
Yeah, I suppose that's fine then since when you hover over `[1]` or `[0]` you can see the type, it's just not as slick/obvious to me as seeing it spelled out and requires some squinting at (maybe it's just me)